darwin/utils/utils.py

```diff
@@ -255,8 +258,12 @@ def find_files(
             raise UnsupportedFileType(path)
 
     files_to_exclude_full_paths = [str(Path(f)) for f in files_to_exclude]
-
-    return [f for f in found_files if str(f) not in files_to_exclude_full_paths]
+    filtered_files = [
+        f for f in found_files if str(f) not in files_to_exclude_full_paths
+    ]
+    if sort:
+        return natsorted(filtered_files)
+    return filtered_files
```

Comment:
We do this because when uploading multi-file items, the order of the files is important. Natural sorting seems the more appropriate approach to me